src/main/java/com/iabtcf/LengthOffsetCache.java

```diff
@@ -0,0 +1,57 @@
+package com.iabtcf;
+
+/*-
+ * #%L
+ * IAB TCF Core Library
+ * %%
+ * Copyright (C) 2020 IAB Technology Laboratory, Inc
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.util.EnumMap;
+import java.util.function.Function;
+
+class LengthOffsetCache {
+    private final ByteBitVector bbv;
+    private final EnumMap<FieldDefs, Integer> lengthCache = new EnumMap<>(FieldDefs.class);
+    private final EnumMap<FieldDefs, Integer> offsetCache = new EnumMap<>(FieldDefs.class);
+
+    public LengthOffsetCache(ByteBitVector bbv) {
+        this.bbv = bbv;
+    }
+
+    public Integer getLength(FieldDefs field, Function<ByteBitVector, Integer> f) {
```

Comment:
is it possible to restructure this code so that we avoid boxing to Integer? I guess we gonna pay a performance hit due to boxing here

Reply:
great question... maybe we can look into using fastutils? http://fastutil.di.unimi.it/

the reason for the Integer is that getLength uses EnumMap to catch the lengths.

Reply:
We could also just implement an EnumIntMap.

Reply:
i created an issue to track this.